nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoCsrfFilter.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Florent Guillaume
+ */
+package org.nuxeo.ecm.platform.web.common.requestcontroller.filter;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Objects;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.platform.web.common.vh.VirtualHostHelper;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+
+/**
+ * Nuxeo CSRF filter, preventing CSRF attacks by rejecting dubious requests.
+ *
+ * @since 10.1
+ */
+public class NuxeoCsrfFilter implements Filter {
+
+    private static final Log log = LogFactory.getLog(NuxeoCsrfFilter.class);
+
+    /**
+     * This configuration property can be set to "true" to disable CSRF protection.
+     */
+    public static final String CSRF_PROTECTION_DISABLED_PROP = "nuxeo.csrf.protection.disabled";
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        // nothing to do
+    }
+
+    @Override
+    public void destroy() {
+        // nothing to do
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
+        if (configurationService != null && configurationService.isBooleanPropertyTrue(CSRF_PROTECTION_DISABLED_PROP)) {
+            // disabled by configuration
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String method = request.getMethod();
+        if ("GET".equals(method) || "HEAD".equals(method) || "OPTIONS".equals(method) || "TRACE".equals(method)) {
```

Comment:
Please can you use constants?  I see HttpServlet has them but they are private.

Reply:
We can either use `javax.ws.rs.HttpMethod` or `org.apache.http.client.methods.HttpGet.METHOD_NAME`, `org.apache.http.client.methods.HttpPost.METHOD_NAME`, etc...